packages/core/src/transform/template/template-to-typescript.ts

```diff
@@ -20,6 +20,22 @@ export type TemplateToTypescriptOptions = {
   specialForms?: Record<string, GlintSpecialForm>;
 };
 
+/**
+ * NOTE: this is tech debt. Tho solving it will require more work than polyfilling old behavior of path.parts
```

Comment:
why / what specifically is tech debt here?

Reply:
I'm adapting an old behavior for the sake of easier PR review, rather than inlining needed changes, and propagating the impact of those changes.

Reply:
ok sounds good